internal/api/core/routing.go

```diff
@@ -272,50 +227,51 @@ func allocateAPIRequest(endpoint string, apiHandler ValidAPIHandler) (any, *APIE
 }
 
 // Reflexively ensure that the api handler is of the correct type/format (e.g. looks like APIHandler).
+// Returns the handler, the type of the input struct, the type of the first output struct, and an error.
```

Comment:
"the type of the input struct" and "the type of the first output struct" sounds a bit awkward (especially that "first" (which implies there is a second?)).
Once again, you can just make it a bit more clear by simplifying it.
Maybe something like "the type of the handler's input, the type of the hander's output".